file_paths.md

```diff
@@ -0,0 +1,108 @@
+        ssl                               # :ssldir                      /etc/puppet/ssl
+
+    /opt/puppetlabs/bin                   # symlink targets of puppet related binaries, excluding ruby, openssl
+        cfacter@                          -> /opt/puppetlabs/puppet/bin/cfacter
+        facter@                           -> /opt/puppetlabs/puppet/bin/facter
+        hiera@                            -> /opt/puppetlabs/puppet/bin/hiera
+        mco@                              -> /opt/puppetlabs/puppet/bin/mco
+        mcollectived@                     -> /opt/puppetlabs/puppet/bin/mco
+        puppet@                           -> /opt/puppetlabs/puppet/bin/puppet
+
+    /opt/puppetlabs/mcollective/
+        plugins
+
+    /opt/puppetlabs/puppet                # ruby-puppet root
+        bin
+            cfacter
+            facter
+            gem
+            hiera
+            mco
+            mcollectived
+            openssl
+            puppet
+            ruby
+        cache                             # :vardir                      /var/lib/puppet
+            bucket                        # :bucketdir
+            client_yaml                   # :clientyamldir
+            client_data                   # :client_datadir
+            clientbucket                  # :clientbucketdir
+            devicedir                     # :devices
+            facts.d                       # :pluginfactdest
+            lib                           # :libdir
+            facts                         # used to generate :factpath
+            puppet-module                 # :module_working_dir
+            reports                       # :reportdir
+            server_datadir                # :server_data
+            state                         # :statedir
+            yaml                          # :yamldir
+        include
+        lib
+        modules                           # system modulepath            /usr/share/puppet/modules
+        share
+            augeas
+            man
+            vim
+        ssl
+
+    /var/log/puppetlabs                   # :logdir                      /var/lib/puppet/log
+        puppet.log                        # :pidfile (depends on run_mode)
+        mcollective.log
+        mcollectived.log
+
+    /var/run/puppetlabs                   # :rundir                      /var/lib/puppet/run
+        agent.pid                         # :pidfile (depends on run_mode)
+        mcollective.pid
+        mcollectived.pid
+
+These sections describe other Puppet packages that rely on puppet-agent to create the initial directory layout.
+
+puppetdb
+--------
+
+    /etc/puppetlabs/puppetdb
+        puppetdb.conf
+
+
+puppet-server
+-------------
+
+    /etc/puppetlabs/puppetserver
+        logback.xml
+        conf.d
+            puppetserver.conf
+
+    /opt/puppetlabs/server                # serverside apps live underneath
+        puppetserver                      # PUP-3089 conflicts with SERVER-357
```

Comment:
{quote}
@shrug could you provide an example of server binaries that exist in /opt/puppetlabs/server/apps/puppetserver/bin and are symlinked into /opt/puppetlabs/server/bin and /opt/puppetlabs/bin? I'm assuming puppetmaster will be?{quote}

Yep, puppetmaster is a good example. 

We could link *all* of the binaries to /opt/puppet/bin, but it seems like it makes sense to leave less-commonly used things (like the httpd binaries) in /opt/puppetlabs/server/bin. Things that are typically interacted with via the init system instead of directly, for example.